Core/Meta/WellKnownTypes.cs

```diff
@@ -100,6 +102,57 @@ public enum BaseType
         Date,
     }
 
+    public static class BaseTypeHelpers
+    {
+        public static BigInteger MinimumInteger(BaseType type)
+        {
+            switch (type)
+            {
+                case BaseType.Byte:
+                case BaseType.UInt16:
+                case BaseType.UInt32:
+                case BaseType.UInt64:
+                    return 0;
+                case BaseType.Int16:
+                    return -(BigInteger.One << 15);
+                case BaseType.Int32:
+                    return -(BigInteger.One << 31);
+                case BaseType.Int64:
+                    return -(BigInteger.One << 63);
+                default:
+                    throw new ArgumentException("MinimumInteger: non-integer type");
+            }
+        }
+
+        public static BigInteger MaximumInteger(BaseType type)
+        {
+            switch (type)
+            {
+                case BaseType.Byte:
+                    return (BigInteger.One << 8) - 1;
+                case BaseType.UInt16:
+                    return (BigInteger.One << 16) - 1;
+                case BaseType.UInt32:
+                    return (BigInteger.One << 32) - 1;
+                case BaseType.UInt64:
+                    return (BigInteger.One << 64) - 1;
+                case BaseType.Int16:
+                    return (BigInteger.One << 15) - 1;
+                case BaseType.Int32:
+                    return (BigInteger.One << 31) - 1;
+                case BaseType.Int64:
+                    return (BigInteger.One << 63) - 1;
+                default:
+                    throw new ArgumentException("MaximumInteger: non-integer type");
+            }
+        }
+
+        public static bool InRange(BaseType type, BigInteger value)
```

Comment:
`CanRepresent`? Could be extended later for other checks like with floats. Just a thought.

Reply:
That's a nicer name! Especially as an extension method: `type.CanRepresent(value)`.